index.bs

```diff
@@ -371,28 +403,164 @@ model]]:
+
+<div algorithm>
+
+To <dfn>process a network cookie access for bounce tracking mitigations</dfn>
+given a [=request=] |request|, perform the following steps:
+
+1. Let |origin| be |request|'s [=request/origin=].
+1. If |origin| is an [=opaque origin=], then abort these steps.
+1. If |request|'s [=request/destination=] is "`document`", then:
+    1. Let |topLevelTraversable| be |request|'s [=request/client=]'s
+        [=environment/target browsing context=]'s
+        [=browsing context/top-level traversable=].
+    1. [=Assert=]: |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=] is not null.
+    1. Let |site| be the result of running [=obtain a site=] given |origin|.
+    1. [=set/Append=] |site|'s [=host=] to |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=]'s
+        [=bounce tracking record/storage access set=].
+
+Issue: TODO: Handle subresource requests and iframes for client-side redirects.
+
+</div>
+
+Note: We currently don't treat cookie reads as stateful, but this would be a
+reasonable future change. We could run [=process a network cookie access for bounce tracking mitigations=]
+in the <a spec="fetch">HTTP-network-or-cache fetch</a> algorithm after step 8.21.1.2,
+"... [=append=] (`Cookie`, cookies) to httpRequest’s [=header list=]. ..."
+
+<h5 id="bounce-tracking-mitigations-nav-start-monkey-patch">Start Navigation
+Monkey Patch</h5>
+
+At the start of a navigation, either initialize a new [=bounce tracking record=],
+or append a client-side redirect to the current [=bounce tracking record=].
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm before
+step 8,
+"[Navigate to a fragment](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate-fragid) given navigable, url, historyHandling, and navigationId."
+
+1. Run [=process navigation start for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>sourceDocument</var>.
+
+<div algorithm>
+
+To <dfn>process navigation start for bounce tracking</dfn> given a [=navigable=]
+|navigable| and [=Document=] |sourceDocument|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. Let |origin| be |sourceDocument|'s [=Document/origin=].
+1. If |origin| is an [=opaque origin=], set |initialHost| to empty string.
+1. Otherwise,
+    1. Let |site| be the result of running [=obtain a site=] given |origin|.
+    1. Set |initialHost| to |site|'s [=host=].
+1. If |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] is false:
+    1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new
+        [=bounce tracking record=] with [=bounce tracking record/initial host=]
+        set to |initialHost|.
+1. Otherwise,
+    1. Let |isUserInitiated| denote whether this navigation was initiated by the user.
```

Comment:
Can we get at the `[=request=]` for this navigation?  The formal way to do this is to look for the `[=request/user-activation=]` attribute on the request:

https://fetch.spec.whatwg.org/#request-user-activation

Its also possible this is already accessed in html navigate algorithm and we could just pass the boolean through to here.

Reply:
Thanks for the pointer! This looks correct.

request isn't available yet at the start of the navigation, but it's populated later using sourceSnapshotParams.hasTransientActivation (https://html.spec.whatwg.org/multipage/browsing-the-web.html#create-navigation-params-by-fetching) so I'm using that flag.